### PR TITLE
[IMP] while clean the odoo module version archive the product

### PR DESCRIPTION
**Pull request by bizzappdev**
in reference to https://github.com/OCA/interface-github/pull/30
when the Odoo module version will be deleted it will archive the related product variant.
If there will be no other product related to that product template then it will archive and unpublish the product template.

### CODE REVIEW

**Comment by bizzappdev** [comment]
@RoelAdriaans  @oscarolar  @StephanRozendaal  Please review.
cc @elicoidal

File: apps_product_creator/models/odoo_module_veresion.py
Comment (elicoidal) [inline apps_product_creator/models/odoo_module_veresion.py]: variant

File: apps_product_creator/models/odoo_module_veresion.py
Comment (elicoidal) [inline apps_product_creator/models/odoo_module_veresion.py]: add your name

**Review by elicoidal — APPROVED** [review]
LGTM

**Comment by legalsylvain** [comment]
there is a typo in the name of the file 
apps_product_creator/models/odoo_module_veresion.py 

-> odoo_module_version.py

**Comment by bizzappdev** [comment]
@elicoidal  @legalsylvain  thanks for your comments. changes are done as suggested. sorry for typos

**Comment by bizzappdev** [comment]
@oscarolar  @RoelAdriaans  @StephanRozendaal @legalsylvain can you please review this PR.

**Review by legalsylvain — APPROVED** [review]
LGTM. Thanks for this work.
(code review / no test).

**Comment by bizzappdev** [comment]
once this PR and https://github.com/OCA/interface-github/pull/30 both are merged. I will test on the migration database for the process to fix old products. and then we can proceed for the update on the production.
cc @elicoidal

**Review by elicoidal — APPROVED** [review]
LGTM

**Comment by elicoidal** [comment]
@oscarolar @RoelAdriaans @StephanRozendaal if you have the opportunity to review?

**Comment by bizzappdev** [comment]
@elicoidal  we have already 2 approval e can go ahead if we want.
NOTE: first I have to test on migration the steps to fix things on production.
that I can do after merge of both PRs

File: apps_product_creator/models/odoo_module_version.py
Comment (oscarolar) [inline apps_product_creator/models/odoo_module_version.py]: Since you are making 3 iterations(`all`, `for`, `mapped`) here I think you should use something with less performance impact:
```python
active_variants = product_obj.search_count(
    [('product_tmpl_id', '=', product.product_tmpl_id.id),
     ('active', '=', True)])
if not active_variants:
    product.product_tmpl_id.write({'active': False, 'website_published': False})
```
what do you think?

Comment (elicoidal) [reply apps_product_creator/models/odoo_module_version.py]: @bizzappdev any feedback on this?

Comment (bizzappdev) [reply apps_product_creator/models/odoo_module_version.py]: @oscarolar thank you for your suggestion, You are right, I have changed the code as per your suggestion.
cc @elicoidal

Comment (elicoidal) [reply apps_product_creator/models/odoo_module_version.py]: I do not see the change here. Did I miss somethign?

Comment (bizzappdev) [reply apps_product_creator/models/odoo_module_version.py]: something strange with github this still shows the outdated changes. 
https://github.com/OCA/apps-store/pull/36/files#diff-87908d4d380713ee148592d4aa5e5069R16
please find the changes it is there already

**Review by elicoidal — APPROVED** [review]
LGTM

**Comment by elicoidal** [comment]
@oscarolar please have a look but I think your remark was attended

**Comment by oscarolar** [comment]
👍 @bizzappdev @elicoidal

**Review by oscarolar — APPROVED** [review]
Good to go.

**Comment by elicoidal** [comment]
@gurneyalex The OCA/interface-github + OCA/apps-store repository should be refreshed in our ERP (not urgent)

**Comment by bizzappdev** [comment]
@elicoidal  @gurneyalex  yes code can be pulled on the server. 
for fixing old products/apps I will provide steps by today or tomorrow evening.

**Comment by elicoidal** [comment]
@bizzappdev could you have a look at Travis too because something got broken in the process.

**Comment by bizzappdev** [comment]
@elicoidal https://travis-ci.org/OCA/apps-store/builds/446015225?utm_source=github_status&utm_medium=notification
seems proper. not sure where you are pointing.

**Comment by elicoidal** [comment]
https://travis-ci.org/OCA/apps-store/builds/446534958?utm_medium=notification&utm_source=email

I am not sure why transifex is failing though: not sure it is important

**Comment by elicoidal** [comment]
@sbidoul maybe can help

**Comment by bizzappdev** [comment]
@elicoidal  it is failing for `travis_makepot` not sure what and why 
@sbidoul  help needed

**Comment by sbidoul** [comment]
makepot failed to push to github because in the meantime another commit (the readme update) was made.

But that commit triggered a new travis build which itself succeeded: https://github.com/OCA/apps-store/commits/11.0

So all is well.

**Comment by sbidoul** [comment]
Shall I install this on odoo-community.org?

**Comment by elicoidal** [comment]
@sbidoul yes please :smile:

**Comment by bizzappdev** [comment]
@sbidoul @gurneyalex  along with the update on the odoo-community.org.

we have to update the latest code for interface-github repo also as https://github.com/OCA/interface-github/pull/30 as the current PR is depending upon this
we have to fix the old products.
for that we have to execute a server action for the same
Please find the detail of the server action as below.
Detail of server action
Model: product.template
Action To Do	Execute Python Code
Python code.

```
for rec in record:
    if len(rec.attribute_line_ids) != 1:
        continue
    att_list = []
    blank_list = []
    for att in rec.attribute_line_ids:
        if not att.value_ids:
            continue
        att_list.append([1, att.id, {'value_ids': [[6, 0, att.value_ids.ids]]}])
        blank_list.append([1, att.id, {'value_ids': [[6, 0, []]]}])
    
    rec.write({'attribute_line_ids':blank_list})
    rec.write({'attribute_line_ids':att_list})
```

As there are missing product variant and odoo module version.
when we will change the value attribute values of the template it will recalculate the odoo module version. in the end, we will have proper products/apps with its module information.

**Comment by sbidoul** [comment]
Installed. The above script is running right now.

**Comment by sbidoul** [comment]
Still running...

**Comment by elicoidal** [comment]
@bizzappdev thanks for the detailed procedure.
I suspect it might take some time as there are about 10K products

**Comment by sbidoul** [comment]
It's done. Can you please check everything is ok?

**Comment by bizzappdev** [comment]
No it is not fixed.. not sure qhy. Do you think we should have screen sharing session or something tomorrow.

**Comment by sbidoul** [comment]
Hm, perhaps because I forgot to upgrade interface-github. I'm relaunching the script.

**Pull request merged**
